site/src/pages/OrganizationSettingsPage/OrganizationProvisionerJobsPage/OrganizationProvisionerJobsPage.stories.tsx

```diff
@@ -0,0 +1,1628 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { expect, userEvent, waitFor, within } from "@storybook/test";
+import type { ProvisionerJob } from "api/typesGenerated";
+import {
+	MockOrganization,
+	MockOrganizationPermissions,
+} from "testHelpers/entities";
+import { daysAgo } from "utils/time";
+import OrganizationProvisionerJobsPage from "./OrganizationProvisionerJobsPage";
+
+const defaultOrganizationSettingsValue = {
+	organization: MockOrganization,
+	organizationPermissionsByOrganizationId: {},
+	organizations: [MockOrganization],
+	organizationPermissions: MockOrganizationPermissions,
+};
+
+const meta: Meta<typeof OrganizationProvisionerJobsPage> = {
+	title: "pages/OrganizationProvisionerJobsPage",
+	component: OrganizationProvisionerJobsPage,
+	args: {
+		defaultOrganizationSettingsValue,
+		getProvisionerJobs: async () => MockProvisionerJobs,
+	},
+};
+
+export default meta;
+type Story = StoryObj<typeof OrganizationProvisionerJobsPage>;
+
+export const Default: Story = {};
+
+export const OrganizationNotFound: Story = {
+	args: {
+		defaultOrganizationSettingsValue: {
+			...defaultOrganizationSettingsValue,
+			organization: undefined,
+		},
+	},
+};
+
+export const Loading: Story = {
+	args: {
+		getProvisionerJobs: () =>
+			new Promise((res) => {
+				setTimeout(res, 100_000);
+			}),
+	},
+};
+
+export const LoadingError: Story = {
+	args: {
+		getProvisionerJobs: async () => {
+			throw new Error("Failed to load jobs");
+		},
+	},
+};
+
+export const RetryAfterError: Story = {
+	args: {
+		getProvisionerJobs: (() => {
+			let count = 0;
+
+			return async () => {
+				count++;
+
+				if (count === 1) {
+					throw new Error("Failed to load jobs");
+				}
+
+				return MockProvisionerJobs;
+			};
+		})(),
+	},
+	play: async ({ canvasElement }) => {
+		const canvas = within(canvasElement);
+		const retryButton = await canvas.findByRole("button", { name: "Retry" });
+
+		userEvent.click(retryButton);
+
+		await waitFor(() => {
+			const rows = canvasElement.querySelectorAll("tbody > tr");
+			expect(rows).toHaveLength(MockProvisionerJobs.length);
+		});
+	},
+};
+
+export const Empty: Story = {
+	args: {
+		getProvisionerJobs: async () => [],
+	},
+};
+
+const MockProvisionerJobs: ProvisionerJob[] = [
```

Comment:
- this should go in testHelpers/entities
- it should use the `MockProvisionerJob` as a baseline, so that it doesn't take up over 1,500 lines of code
- idk exactly what makes each value different, but you could probably do something like `Array.from({ length: 50 }, (_, i) => ({ ...MockProvisionerJob, otherThings: ... }))` to make those differences a lot clearer

Reply:
I just like to use data returned from the API to test with "real data" but I see value on what you are saying... 🤔

Reply:
as long as it's got the type assertion and is based on actual api data, I think it's still real enough. and like we've been talking about, we have the e2e tests as the definitive "backend + frontend work good" tests. 😄